aries_cloudagent/resolver/base.py

```diff
@@ -75,16 +97,19 @@ def supports(self, did: Union[str, DID]) -> bool:
 
         return False
 
-    async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
+    async def resolve(self, profile: Profile, did: Union[str, DID]) -> Resolution:
         """Resolve a DID using this resolver."""
         py_did = DID(did) if isinstance(did, str) else did
 
         if not self.supports(py_did):
             raise DIDMethodNotSupported(
                 f"{self.__class__.__name__} does not support DID method {py_did.method}"
             )
-
+        previous_time = datetime.utcnow()
         did_document = await self._resolve(profile, str(py_did))
+        resolver_metadata = await self._retrieve_resolver_metadata(
```

Comment:
Do we see the `_retrieve_resolver_metadata` method being used elsewhere? If not, I think it might be clearer to just do the body of the method directly here instead of creating a separate method.

Reply:
I think we should keep it because the resolver function is quite large currently. And with this function, we can encapsulate this metadata built. I think it's more readable in this way.